samples/OpenIdConnectSample/Startup.cs

```diff
@@ -39,6 +40,23 @@ public void Configure(IApplicationBuilder app, ILoggerFactory loggerfactory)
 
             app.Run(async context =>
             {
+                var authContext = new AuthenticateContext(CookieAuthenticationDefaults.AuthenticationScheme);
+                await context.Authentication.AuthenticateAsync(authContext);
+                if (context.Request.Path == new PathString("/remote-sign-out"))
+                {
+                    var session = authContext.Properties[OpenIdConnectSessionProperties.SessionState];
+                    var sid = context.Request.Query["sid"];
+                    if (string.Equals(sid, session, System.StringComparison.Ordinal))
```

Comment:
If the 'sid' claim is not there, it is not being sent. 
The only session AAD sends is the one attached with the signin response, which is used for RP initiated signout. I suspect it should work with AAD, but in general will not for all IDP's.

The correct thing is the 'sid' claim which should be available if logout_session_supported is true which it part of discovery.